hs_composite_resource/tests/test_composite_resource.py

```diff
@@ -91,6 +94,70 @@ def test_create_composite_resource_with_file_upload(self):
         # deleted the resource already
         self.composite_resource = None
 
+    def test_add_and_edit_referened_url(self):
+        # test that referenced url can be added to composite resource as a genericlogical file
+        # type single file aggregation, and can also be edited
+
+        # there should not be any resource at this point
+        self.assertEqual(BaseResource.objects.count(), 0)
+
+        # test invalid url fails to be added to an empty composite resource
+        self.create_composite_resource()
+        url_file_base_name = 'test_url'
+        ret_status, msg = add_reference_url_to_resource(self.user,
+                                                        self.composite_resource.short_id,
+                                                        self.invalid_url, url_file_base_name,
+                                                        'data/contents')
+        self.assertEqual(ret_status, status.HTTP_400_BAD_REQUEST)
```

Comment:
What is the message?  I think it's worth asserting the msg says something about the url being invalid.

Reply:
@sblack-usu Good point. Have added messages for all assert failures.